library/data/tags/src/main/java/org/quiltmc/qsl/tag/api/TagType.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.tag.api;
+
+/**
+ * Represents tag types. Tag types define how tags are loaded and synced.
+ *
+ * @see org.quiltmc.qsl.tag.api
+ */
+public enum TagType {
+	/**
+	 * Represents a tag type whose tags are required to start but not to connect, they will sync to the client.
+	 */
+	SERVER_REQUIRED(true),
+	/**
+	 * Represents a tag type whose tags are required to start and to connect, they will sync if present on the server.
+	 * <p>
+	 * If the client has tags which are of this type, and the server does not,
+	 * the client will disconnect when attempting to connect to said server.
+	 */
+	CLIENT_SERVER_REQUIRED(true),
+	/**
+	 * Represents a neutral tag type, tags are not required to start nor to connect,
+	 * but they will sync if present on the server.
+	 * <p>
+	 * The client does not provide a default.
+	 */
+	NEUTRAL(false),
```

Comment:
- `SERVER_REQUIRED` means it's a tag that is only required for a server to start (or load a singleplayer world), 
- `CLIENT_SERVER_REQUIRED` means that in addition to being required for a server to start, the client also must receive that tag from the server when connecting (default behavior of most tags in MC),
- `NEUTRAL` means it's a tag that will be loaded and synced, but will not be required for server start or for client connection, if the server doesn't have it, the client won't have it either. This is the current behavior in Fabric API.
- `CLIENT_SERVER_SYNC` means it's mostly a client tag, not required to start nor connect, if the server has the tag it will sync it to the client, otherwise the client will use the defaults (loading from the data directory still)
- `CLIENT_ONLY` means the tag only exists on the client, located in the `assets` directory instead.

Reply:
`SERVER_REQUIRED` is fine.

`CLIENT_SERVER_REQUIRED` -> `REQUIRED`. The game will not function if the server and client do not have an instance of the tag. `REQUIRED` communicates this idea succinctly.

`NEUTRAL` -> `OPTIONAL`. The client or server does not require a tag. The server is the authoritative source of the tag

`CLIENT_SERVER_SYNC` -> `CLIENT_FALLBACK`. Think of it like an `OPTIONAL` that uses the client's tag as a fallback. The server is considered to be authoritative with it's own tag list.

`CLIENT_ONLY` is fine.